flink-connector-aws/flink-connector-dynamodb/src/main/java/org/apache/flink/connector/dynamodb/sink/DynamoDbTypeInformedElementConverter.java

```diff
@@ -0,0 +1,245 @@
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * A {@link ElementConverter} that converts an element to a {@link DynamoDbWriteRequest} using
+ * TypeInformation provided.
+ */
+@PublicEvolving
+public class DynamoDbTypeInformedElementConverter<inputT>
+        implements ElementConverter<inputT, DynamoDbWriteRequest> {
+    private final CompositeType<inputT> typeInfo;
+
+    /**
+     * Creates a {@link DynamoDbTypeInformedElementConverter} that converts an element to a {@link
+     * DynamoDbWriteRequest} using the provided {@link CompositeType}. Usage: {@code new
+     * DynamoDbTypeInformedElementConverter<>(TypeInformation.of(MyPojoClass.class))}
+     *
+     * @param typeInfo The {@link CompositeType} that provides the type information for the element.
+     */
+    public DynamoDbTypeInformedElementConverter(CompositeType<inputT> typeInfo) {
+        this.typeInfo = typeInfo;
+    }
+
+    @Override
+    public DynamoDbWriteRequest apply(inputT input, SinkWriter.Context context) {
+        try {
+            return DynamoDbWriteRequest.builder()
+                    .setType(DynamoDbWriteRequestType.PUT)
+                    .setItem(convertElementUsingTypeInfo(input, typeInfo))
+                    .build();
+        } catch (IllegalArgumentException e) {
+            throw new FlinkRuntimeException("Couldn't convert Element to AttributeVal", e);
+        }
+    }
+
+    private <attT> Map<String, AttributeValue> convertElementUsingTypeInfo(
+            attT t, CompositeType<attT> typeInfo) {
+        Map<String, AttributeValue> map = new HashMap<>();
+        for (String fieldKey : typeInfo.getFieldNames()) {
+            TypeInformation<attT> fieldType = typeInfo.getTypeAt(fieldKey);
+            try {
+                Field field = t.getClass().getDeclaredField(fieldKey);
+                field.setAccessible(true);
+                Object fieldVal = field.get(t);
+                checkTypeCompatibility(fieldVal, fieldType);
+                attT fieldValCaster = (attT) fieldVal;
+                map.put(fieldKey, convertValue(fieldValCaster, fieldType));
+            } catch (NoSuchFieldException | IllegalAccessException e) {
+                throw new IllegalArgumentException(
+                        String.format(
+                                "Failed to extract field %s declared in TypeInfo "
+                                        + "from Object %s",
+                                fieldKey, t),
+                        e);
+            }
+        }
+
+        return map;
+    }
+
+    private <attrT> AttributeValue convertValue(
+            attrT attribute, TypeInformation<attrT> objectTypeInformation) {
+        if (attribute == null) {
+            return AttributeValue.builder().nul(true).build();
```

Comment:
Not sure this is the correct thing to do here? I believe a common practise is to omit null fields rather than setting to null. Maybe this could be a configuration on the element converter?

Reply:
@dannycranmer , yeah just checked `testConvertOrderToDynamoDbWriteRequestWithIgnoresNull`.